### ISSUE TITLE
FEAT: LightGBM trainer with IC early-stopping and flat-file model registry

### ISSUE THREAD

**Issue by Analyst-Ninja** [Analyst-Ninja/aurum#53]
## Problem / Motivation

With preprocessing and leak-free splits in place, the model itself. Two choices in this issue are load-bearing and easy to get wrong by using library defaults.

**Early stopping on L2 stops on the wrong thing.** L2 on a return target is dominated by the tails — a handful of earnings gaps — while the strategy consumes the *rank ordering*. A model early-stopped on L2 is tuned to predict outliers.

**Default hyperparameters overfit instantly.** Cross-sectional return targets have R² in the low single-digit basis points. LightGBM's defaults are tuned for clean data and will fit noise on 2.7M rows.

Part of EPIC #50 (Phase 6). Depends on #52. Blocks #54, #55, #56.

## What we want to do

**`src/modeling/models/lgbm.py`** — `LGBMRegressor`, L2 objective, on the per-date winsorized and standardized `fwd_ret_5d_excess`.

Why LightGBM, recorded so it can be revisited rather than re-argued:
- **Native NaN routing** — 34.6% of rows have no fundamentals; `roe` is 32.6% null even among rows that do. Imputing zero is wrong (zero is a meaningful margin); imputing the median injects cross-sectional information. LightGBM learns a default direction per split.
- **Native categoricals** for `sector` / `industry`.
- **`shap.TreeExplainer` is exact and fast on trees** — `mart_feature_summary` and the whole selection loop assume tree SHAP. A neural model forces KernelSHAP: approximate, and orders of magnitude slower over 200+ columns.
- Rejected: XGBoost (equivalent; LightGBM is faster here with better categorical support), linear/Ridge (kept as a **baseline**, not the model), neural nets (the data is tabular and small in the way that matters — ~6,600 distinct dates, not 2.9M independent samples), `LGBMRanker` (a secondary head under #54, not the primary).

**Custom eval function** = mean per-date Spearman IC on the validation fold, used for early stopping. Not L2.

**Starting grid**, hand-specified; Optuna is explicitly deferred:

| Param | Range |
|---|---|
| `num_leaves` | 15 – 63 |
| `min_child_samples` | 500 – 5000 (very large — the panel is huge and noisy) |
| `learning_rate` | 0.01 – 0.05 |
| `n_estimators` | up to 3000, early stopping decides |
| `feature_fraction` | 0.3 – 0.6 (heavy collinearity across the `_z`/`_decile`/`_vs_sector` triplets) |
| `bagging_fraction` / `bagging_freq` | 0.7 / 1 |
| `lambda_l2` | 1 – 50 |

Selection uses folds 121–297 only. The 24-month holdout never informs it.

**`src/modeling/models/registry.py`** — flat files, no MLflow (spec §7 Q5 defers it):

```
models/{version}/                 version = {YYYYMMDD}-{git short sha}
├── model.txt                     LightGBM native format, not pickle — survives version drift
├── metadata.json                 git_sha, dbt manifest hash, target, config hash, train/val/test
│                                 windows, fold spec, hyperparams, package versions, row/col counts,
│                                 n_configs_tried
├── feature_manifest.json
├── preprocess_manifest.json
├── metrics.json                  (#54)
├── shap/                         (#55)
└── backtest/                     (#56)
models/latest -> models/{version}/
```

`.gitignore` excludes `models/*/model.txt` and `data/`. `metadata.json`, `metrics.json` and `shap/ranking.csv` **are committed** — the run history stays reviewable in git diffs, the same reasoning that made `selected_features.csv` a seed rather than a table.

## Implementation notes

- `model.txt` (`booster.save_model`) over pickle: a pickled sklearn wrapper breaks across library versions, and this repo will outlive the current lightgbm pin.
- `n_configs_tried` must be recorded in `metadata.json` — a Sharpe selected out of 200 attempts is not the Sharpe you will get, and #56's deflated-Sharpe check reads this field.
- The dbt manifest hash goes in `metadata.json` so #57's structural retrain trigger can detect a warehouse change without re-reading the warehouse.
- Fit ~15 models (one per refit boundary), not 321. Refit cadence is a config knob.

## Acceptance criteria

- [ ] `cli train -c src/modeling/configs/lgbm_xs_excess_5d.yaml` runs end to end and writes a complete `models/{version}/` directory
- [ ] Early stopping fires on mean per-date Spearman IC; a unit test confirms the custom eval matches a `scipy.stats.spearmanr` reference on a small frame
- [ ] `metadata.json` carries git SHA, dbt manifest hash, config hash, exact fold boundaries, package versions and `n_configs_tried`
- [ ] Reloading `model.txt` and predicting reproduces the training-run predictions exactly
- [ ] `models/latest` resolves to the newest promoted version
- [ ] Holdout folds are provably untouched — assert the fitted models' training date ranges all end before the holdout boundary
- [ ] `uv run ruff check src/ main.py` clean

## References

- `docs/modeling/modeling-design.md` §Model choice
- `docs/modeling/training-and-retraining.md` §Hyperparameters, §Registry
- `docs/architecture/TECHNICAL_SPEC.md` §3.7, §7 Q5
- EPIC #50, depends on #52
